### PR TITLE
Fixes for AMIP

### PR DESCRIPTION
**Pull request by LenkaNovak**
closes https://github.com/CliMA/ClimaCoupler.jl/issues/568

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: test/component_model_tests/climaatmos_standalone/atmos_driver.jl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using ClimaComms
```
Comment (juliasloan25) [inline test/component_model_tests/climaatmos_standalone/atmos_driver.jl:1]: Should we set `comms_ctx` like we did in target/atmos_driver.jl before?

File: src/ConservationChecker.jl [hunk 2]
```diff
@@ -260,23 +263,33 @@ function plot_global_conservation(
...
-        label = "tot",
+        log.(rse),
+        label = "rse error",
         xlabel = "time [days]",
-        ylabel = "log( | e(t) - e(t=0)| / e(t=0))",
+        ylabel = "log( |x(t) - x(t=0)| / Σx(t=T) )",
     )
     Plots.savefig(figname2)
+
+    # check that the relative error is small (TODO: reduce this to sqrt(eps(FT)))
```
Comment (juliasloan25) [inline src/ConservationChecker.jl:289]: could we open an issue for this TODO so we remember to fix it? :)

File: experiments/AMIP/components/land/bucket_utils.jl [hunk 3]
```diff
@@ -54,7 +54,7 @@ function update_field!(sim::BucketSimulation, ::Val{:turbulent_energy_flux}, fie
 end
 function update_field!(sim::BucketSimulation, ::Val{:turbulent_moisture_flux}, field)
     ρ_liq = (LSMP.ρ_cloud_liq(sim.model.parameters.earth_param_set))
-    parent(sim.integrator.p.bucket.evaporation) .= parent(field ./ ρ_liq)
+    parent(sim.integrator.p.bucket.evaporation) .= parent(field ./ ρ_liq) # TODO: account for sublimation
```
Comment (juliasloan25) [inline experiments/AMIP/components/land/bucket_utils.jl:57]: can we open an issue for this TODO too?

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl [hunk 4]
```diff
@@ -333,7 +339,22 @@ function get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:F_radiative_TOA})
     end
 end
 
-get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:energy}) = atmos_sim.integrator.u.c.ρe_tot
+function get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:energy})
+    thermo_params = get_thermo_params(atmos_sim)
+
+    ᶜS_ρq_tot = atmos_sim.integrator.p.precipitation.ᶜS_ρq_tot
+    ᶜts = atmos_sim.integrator.p.precomputed.ᶜts
+    ᶜΦ = atmos_sim.integrator.p.core.ᶜΦ
```
Comment (juliasloan25) [inline experiments/AMIP/components/atmosphere/climaatmos_init.jl:347]: I think this can all go in the `if` branch

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl [hunk 5]
```diff
@@ -97,8 +100,11 @@ function get_field(sim::ClimaAtmosSimulation, ::Val{:liquid_precipitation})
     sim.integrator.p.precipitation.col_integrated_rain .* ρ_liq # kg/m^2/s
 end
 function get_field(sim::ClimaAtmosSimulation, ::Val{:snow_precipitation})
-    ρ_liq = CAP.ρ_cloud_liq(sim.integrator.p.params)
-    sim.integrator.p.precipitation.col_integrated_snow .* ρ_liq  # kg/m^2/s
+    # TODO: use this upon Atmos 0.19.3 release
```
Comment (juliasloan25) [inline experiments/AMIP/components/atmosphere/climaatmos_init.jl:103]: it will be v0.20 not 0.19.3 (but same idea) :)

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl [hunk 6]
```diff
@@ -81,6 +81,9 @@ function atmos_init(::Type{FT}, atmos_config_dict::Dict) where {FT}
     integrator.p.precomputed.sfc_conditions.ρ_flux_uₕ.components .= Ref(SMatrix{1, 2}([FT(0), FT(0)]))
     integrator.p.precipitation.col_integrated_rain .= FT(0)
     integrator.p.precipitation.col_integrated_snow .= FT(0)
+    integrator.p.precipitation.ᶜS_ρq_tot .= FT(0)
+    integrator.p.precipitation.ᶜ3d_rain .= FT(0)
+    integrator.p.precipitation.ᶜ3d_snow .= FT(0)
```
Comment (juliasloan25) [inline experiments/AMIP/components/atmosphere/climaatmos_init.jl:86]: are these new fields in the atmos integrator? or were we just not using them before?

File: experiments/AMIP/Manifest.toml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.10.0"
+julia_version = "1.9.3"
```
Comment (juliasloan25) [inline experiments/AMIP/Manifest.toml:3]: we don't want to downgrade to julia v1.9.3 - could you upgrade to 1.10 then resolve the manifest?

File: config/longrun_configs/amip_target.yml [hunk 8]
```diff
@@ -1,18 +1,23 @@
-run_name: "amip_target"
 anim: true
```
Comment (juliasloan25) [inline config/longrun_configs/amip_target.yml:1]: the alphabetization is really nice!! it would be nice to have this standardized across all the yml files too (but we can do that in a different PR)

**Review by juliasloan25 — COMMENTED** [review]
Looks good! I left a few comments

File: config/longrun_configs/amip_target_topo.yml [hunk 9]
```diff
@@ -1,21 +1,27 @@
...
-topography: "Earth"
+surface_setup: "PrescribedSurface"
+t_end: "200days"
 topo_smoothing: true
-use_reference_state: false
\ No newline at end of file
+topography: "Earth"
+turb_flux_partition: "CombinedStateFluxes"
+use_reference_state: false
+vert_diff: "true"
```
Comment (szy21) [inline config/longrun_configs/amip_target_topo.yml:27]: Same here, is the moist Held-Suarez diffusion intended?

File: test/component_model_tests/climaatmos_standalone/atmos_driver.jl (hunk 1 above)
Comment (LenkaNovak) [reply test/component_model_tests/climaatmos_standalone/atmos_driver.jl:1]: I don't think we need it here, since we're just running ClimaAtmos on its own. It's basically doing [this](https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/hybrid/driver.jl#L6-L16), but it's useful to have it in the Coupler for performance and release testing.

File: experiments/AMIP/Manifest.toml (hunk 7 above)
Comment (LenkaNovak) [reply experiments/AMIP/Manifest.toml:3]: Ahh bad rebase! Good catch!!

**Review by szy21 — COMMENTED** [review]
Just a few comments on the config file.

File: src/ConservationChecker.jl (hunk 2 above)
Comment (LenkaNovak) [reply src/ConservationChecker.jl:289]: Done https://github.com/CliMA/ClimaCoupler.jl/issues/594

File: experiments/AMIP/components/land/bucket_utils.jl (hunk 3 above)
Comment (LenkaNovak) [reply experiments/AMIP/components/land/bucket_utils.jl:57]: same issue as above https://github.com/CliMA/ClimaCoupler.jl/issues/594

File: config/longrun_configs/amip_target.yml [hunk 10]
```diff
@@ -1,18 +1,23 @@
...
-dt_save_state_to_disk: "20days"
-apply_limiter: false
-hourly_checkpoint: true
 turb_flux_partition: "CombinedStateFluxes"
-atmos_config_file: "config/longrun_configs/longrun_aquaplanet_rhoe_equil_55km_nz63_clearsky_0M.yml"
-surface_setup: "PrescribedSurface"
-land_albedo_type: "map_temporal"
-start_date: "19790301"
\ No newline at end of file
+vert_diff: "true"
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply config/longrun_configs/amip_target.yml:23]: I am waiting for the long runs to finish. I will pick the more stable one ;-)

File: test/component_model_tests/climaatmos_standalone/atmos_driver.jl (hunk 1 above)
Comment (juliasloan25) [reply test/component_model_tests/climaatmos_standalone/atmos_driver.jl:1]: I see!

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl (hunk 6 above)
Comment (LenkaNovak) [reply experiments/AMIP/components/atmosphere/climaatmos_init.jl:86]: They are, but they are not initialized to zero when we first call the precipitation energy check (implemented in this PR).

File: config/longrun_configs/amip_target.yml (hunk 8 above)
Comment (LenkaNovak) [reply config/longrun_configs/amip_target.yml:1]: I was going to do this as part of a separate PR right after this, so that I don't overwhelm you guys with too many files to review at once. :)

Comment (LenkaNovak) [reply config/longrun_configs/amip_target.yml:1]: https://github.com/CliMA/ClimaCoupler.jl/issues/595

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl (hunk 5 above)
Comment (LenkaNovak) [reply experiments/AMIP/components/atmosphere/climaatmos_init.jl:103]: Nice! :)

Comment (LenkaNovak) [reply experiments/AMIP/components/atmosphere/climaatmos_init.jl:103]: reminder: https://github.com/CliMA/ClimaCoupler.jl/issues/596

File: config/longrun_configs/amip_target_topo.yml [hunk 11]
```diff
@@ -1,21 +1,27 @@
...
 start_date: "19790301"
-topography: "Earth"
+surface_setup: "PrescribedSurface"
+t_end: "200days"
 topo_smoothing: true
-use_reference_state: false
\ No newline at end of file
+topography: "Earth"
+turb_flux_partition: "CombinedStateFluxes"
+use_reference_state: false
```
[the remark this answers is not in the document]
Comment (LenkaNovak) [reply config/longrun_configs/amip_target_topo.yml:26]: I can try it and rm it if it doesn't break!

**Review by juliasloan25 — APPROVED** [review]
